samples/opentelemetryTracing.js

```diff
@@ -0,0 +1,97 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This sample demonstrates how to add OpenTelemetry tracing to the
+ * Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+'use strict';
+
+// sample-metadata:
+//   title: OpenTelemetry Tracing
+//   description: Demonstrates how to enable OpenTelemetry tracing in
+//     a publisher or subscriber.
+//   usage: node opentelemetryTracing.js <topic-name> <subscription-name>
+
+const SUBSCRIBER_TIMEOUT = 10;
+
+function main(
+  topicName = 'YOUR_TOPIC_NAME',
+  subscriptionName = 'YOUR_SUBSCRIPTION_NAME',
+  data = {foo: 'bar'}
+) {
+  // [START opentelemetry_tracing]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const topicName = 'my-topic';
+  // const subscriptionName = 'my-subscription';
+  // const data = 'Hello, world!";
+
+  // Imports the Google Cloud client library
+  const {PubSub} = require('@google-cloud/pubsub');
+
+  // Imports the OpenTelemetry API
+  const {opentelemetry} = require('@opentelemetry/api');
+
+  // Imports the OpenTelemetry span handlers and exporter
+  const {
+    SimpleSpanProcessor,
+    BasicTracerProvider,
+    ConsoleSpanExporter,
+  } = require('@opentelemetry/tracing');
+
+  // Set up span processing and specify the console as the span exporter
+  const provider = new BasicTracerProvider();
+  const exporter = new ConsoleSpanExporter();
+  provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+
+  provider.register();
+  opentelemetry.trace.setGlobalTracerProvider(provider);
+
+  // Creates a client; cache this for further use
+  const pubSubClient = new PubSub();
+
+  async function publishMessage() {
+    // Publishes the message as a string, e.g. "Hello, world!" or JSON.stringify(someObject)
+    const dataBuffer = Buffer.from(data);
+
+    const messageId = await pubSubClient.topic(topicName).publish(dataBuffer);
+    console.log(`Message ${messageId} published.`);
+  }
+
+  async function subscriptionListen() {
+    // Message handler for subscriber
+    const messageHandler = message => {
+      console.log(`Message ${message.id} received.`);
+      message.ack();
+    };
+
+    // Listens for new messages from the topic
+    pubSubClient.subscription(subscriptionName).on('message', messageHandler);
+    setTimeout(() => {
+      pubSubClient.subscription(subscriptionName).removeAllListeners();
+    }, SUBSCRIBER_TIMEOUT * 1000);
+  }
+
+  publishMessage().then(subscriptionListen());
```

Comment:
This is missing a doc end tag, though it may not matter at this point if nothing in the docsite is expecting it to be here. (We would need to canonicalize the the tag name anyway.)

Reply:
I want to include a link to this example in the blog post. How should we canonicalize the tag name?

Reply:
Let me see if I can find the right person to ask... there were some changes in that area lately.